### ISSUE TITLE
Configuration is unable to save.

### ISSUE THREAD

**Issue by RecNes** [hassio-addons/addon-nut#395]
# Problem/Motivation

> Configuration is failing to save by using UI or YAML editor.

## Expected behavior

> Configuration should've saved

## Actual behavior

> Nuts is niether starting with current configuration nor saving my configuration changes.
If I cut out big part of the configuration and put empty string to password, at that time configuration can be saved.

## Steps to reproduce

Current version: 0.13.1

> Aplly this configuration and try to save:

```
users:
  - username: monuser
    password: secret
    upsmon: master
    instcmds:
      - all
    actions: []
devices:
  - name: ups
    driver: nutdrv_qx
    port: auto
    config:
      - vendorid = 0001
      - productid = 0000
      - langid_fix = 0x409
      - protocol = megatec
      - subdriver = snr
      - runtimecal = 180,100,540,50
      - desc = "Makelsan Lion 2200VA"
mode: netserver
shutdown_host: false
log_level: error
list_usb_devices: true
i_like_to_be_pwned: true
```

## Proposed changes

> (If you have a proposed change, workaround or fix,
> describe the rationale behind it)

**Comment by RecNes** [comment]
I reset to default but it's unable to save, either.
The default configuration app genereted:

```
users:
  - username: ""
    password: ""
    instcmds:
      - all
    actions: []
devices:
  - name: myups
    driver: usbhid-ups
    port: auto
    config: []
mode: netserver
shutdown_host: "false"
```

**Comment by RecNes** [comment]
I discovered that the problem was caused by an incorrect DNS record. 
More precisely, it was due to my directing HASSIO to the DNS masquerade service I set up locally, which resulted in it being unable to resolve. 

However, I couldn't comment on why it exhibited a behavior like not saving the configuration due to the inability to perform DNS resolution. 

It seems that everything is back to normal after redirecting the DNS settings to an external server.
